### PR TITLE
fix(core/id-headers): move self-links adjacent to headers

### LINKED ISSUE #3898: Overly verbose section permalinks for screen reader users
## Important info

- URL to affected spec: https://w3c.github.io/aria-practices/
- ReSpec version: Unknown; there is no ReSpec button on the linked editor's draft.
- [x] I did a "hard refresh", but it's still busted.

## Description of problem

As highlighted in https://github.com/w3c/aria-practices/issues/2159, the changes in https://github.com/w3c/respec/pull/3874 make all section permalinks, and hence section headings, incredibly verbose for screen reader users.  That issue expands on the problematic behaviour and a suggested fix.

CC @marcoscaceres / @mcking65 / @sinabahram

### PR DESCRIPTION
**Pull request by marcoscaceres**
fixes #3898 

The `a.self-link` are now placed after the headers.

### CODE REVIEW

File: tests/spec/core/id-headers-spec.js [hunk 1]
```diff
@@ -38,28 +38,30 @@ describe("Core - ID headers", () => {
...
-      );
-      const appendix = doc.querySelector("#a1 a.self-link");
-      expect(appendix.getAttribute("aria-label")).toBe(
-        "Permalink for Appendix A."
-      );
-      const deepAppendix = doc.querySelector("#a2 a.self-link");
-      expect(deepAppendix.getAttribute("aria-label")).toBe(
-        "Permalink for Appendix A.1"
-      );
+    it("adds lables sections and appendix", () => {
```
Comment (marcoscaceres) [inline tests/spec/core/id-headers-spec.js:44]:
```suggestion
    it("distinguishes between regular sections and appendixes, including ones without a section number", () => {
```

File: src/core/id-headers.js [hunk 2]
```diff
@@ -14,7 +14,9 @@ const localizationStrings = {
      * @param {"Appendix" | "Section"} sectionType
      */
     permalinkLabel(sectionType, sectionNumber) {
-      let label = `Permalink for ${sectionType}`;
+      let label = `Permalink for${
+        !sectionNumber ? " this" : ""
```
Comment (sidvishnoi) [inline src/core/id-headers.js:18]: Curious why we might not want "this" for all headings.

File: src/core/id-headers.js [hunk 3]
```diff
@@ -41,8 +43,9 @@ export function run(conf) {
       h.closest(".appendix") ? "Appendix" : "Section",
       h.querySelector(":scope > bdi.secno")
     );
-    h.append(html`
-      <a href="${`#${id}`}" class="self-link" aria-label="${label}"></a>
-    `);
+    h.insertAdjacentElement(
+      "afterend",
+      html`<a href="${`#${id}`}" class="self-link" aria-label="${label}"></a>`
```
Comment (sidvishnoi) [inline src/core/id-headers.js:48]: I tested with Orca + Chrome on Linux.
It's announced like:

```
< (Click heading to focus)
> Status of this document heading level 2
< (tab)
> Permalink for this section link Status of this document Permalink for this section
```

Comment (sidvishnoi) [reply src/core/id-headers.js:48]: And it sometimes kinda loops (heard section 4 times in a row)

**Review by sidvishnoi — APPROVED** [review]

File: src/core/id-headers.js (hunk 2 above)
Comment (marcoscaceres) [reply src/core/id-headers.js:18]: Its for a numbered section: "For section X.X", VS for a section without a number: "For **this** Section". It makes sense in context.

**Pull request merged**